src/main/scala/rocket/HellaCacheArbiter.scala

```diff
@@ -45,20 +48,20 @@ class HellaCacheArbiter(n: Int)(implicit p: Parameters) extends Module
         connect_s1()
       } else {
         when (req.valid) { connect_s0() }
-        when (s1_id === UInt(i)) { connect_s1() }
+        when (s1_id === i.U) { connect_s1() }
       }
     }
 
     for (i <- 0 until n) {
       val resp = io.requestor(i).resp
-      val tag_hit = io.mem.resp.bits.tag(log2Up(n)-1,0) === UInt(i)
+      val tag_hit = io.mem.resp.bits.tag(log2Ceil(n)-1,0) === i.U
       resp.valid := io.mem.resp.valid && tag_hit
       io.requestor(i).s2_xcpt := io.mem.s2_xcpt
       io.requestor(i).ordered := io.mem.ordered
       io.requestor(i).perf := io.mem.perf
-      io.requestor(i).s2_nack := io.mem.s2_nack && s2_id === UInt(i)
+      io.requestor(i).s2_nack := io.mem.s2_nack && s2_id === i.U
       resp.bits := io.mem.resp.bits
-      resp.bits.tag := io.mem.resp.bits.tag >> log2Up(n)
+      resp.bits.tag := io.mem.resp.bits.tag >> log2Ceil(n)
```

Comment:
I don’t believe these two log2Up changes are semantically equivalent because n can be 1.